document-sandbox-samples/communication-iframe-documentSandbox/src/index.html

```diff
@@ -76,38 +76,38 @@
                 console.log("AddOnSdk is ready for use.");
 
                 const { runtime } = AddOnSdk.instance;
+                // APIs to be exposed to the document sandbox runtime
+                // i.e., to the `code.js` file of this add-on.
                 const uiApi = {
                     performWorkOnUI: function(data, someFlag) {
                         // Do some ui operation
                     },
                     getDataFromUI: async function() {
-                        let resolver = undefined;
                         const promise = new Promise(resolve => {
-                            resolver = resolve;
+                            resolve("button_color_blue");
                         });
-                        setTimeout(() => {
-                            resolver("button_color_blue");
-                        }, 10);
                         return await promise;
```

Comment:
Similarly... this usage of Promises feels like an uncommon style.

I'd suggest we either strip it way down like this (since no timeout is here anymore):
```js
// Example of returning a user choice from hypothetical UI
return "button_color_blue";
```

Or if we want to demonstrate Promises still for whatever reason, a more typical pattern like this:
```js
return new Promise(resolve => {
    // Example of returning a user choice from hypothetical UI
    resolve("button_color_blue");
});
```